packages/workflows/src/utils/composer/transform.ts

```diff
@@ -13,14 +13,62 @@ type Func1<T extends object | WorkflowData, U> = (
 
 type Func<T, U> = (input: T, context: StepExecutionContext) => U | Promise<U>
 
+/**
+ *
+ * This function transforms the output of other steps using the provided functions.
+ *
+ * This is useful if you're using the value(s) of some step(s) as an input to a later step. As you can't directly manipulate data in the  workflow constructor function passed to {@link createWorkflow},
+ * the `transform` function provides access to the runtime value of the step(s) output so that you can manipulate them.
+ *
+ * This is also useful if you're using the runtime value of some step(s) as the output of a workflow.
+ *
+ * @returns There's no expected value to be returned by the `transform` function.
+ *
+ * @example
+ * import {
+ *   createWorkflow,
+ *   transform
+ * } from "@medusajs/workflows"
+ * import { step1, step2 } from "./steps"
+ *
+ * type WorkflowInput = {
+ *   name: string
+ * }
+ *
+ * type WorkflowOutput = {
+ *   message: string
+ * }
+ *
+ * const myWorkflow = createWorkflow<WorkflowInput, WorkflowOutput>
+ *   ("hello-world", (input) => {
+ *     const str1 = step1(input)
+ *     const str2 = step2(input)
+ *
+ *     return transform({
+ *       str1,
+ *       str2
+ *     }, (input) => ({
+ *       message: `${input.str1}${input.str2}`
+ *     }))
+ * })
+ */
 // prettier-ignore
 // eslint-disable-next-line max-len
 export function transform<T extends object | WorkflowData, RFinal>(
+  /**
+   * The output(s) of other step functions.
+   */
   values: T,
+  /**
+   * The transform function(s) used to perform action on the runtime values of the provided `values`.
```

Comment:
maybe remove the s of function so that we constraint people to use a single function in most cases except for advanced users